server/fleet/datastore.go

```diff
@@ -1653,6 +1653,11 @@ type Datastore interface {
 	// (if set) post-install scripts, otherwise those fields are left empty.
 	GetSoftwareInstallerMetadataByTeamAndTitleID(ctx context.Context, teamID *uint, titleID uint, withScriptContents bool) (*SoftwareInstaller, error)
 
+	CancelPendingInstallsForInstallerID(ctx context.Context, id uint) error
+	HideExistingInstallCountsForInstallerID(ctx context.Context, id uint) error
+	SaveInstallerUpdates(ctx context.Context, payload *UpdateSoftwareInstallerPayload) error
+	UpdateInstallerSelfServiceFlag(ctx context.Context, selfService bool, id uint) error
```

Comment:
could use some comment docs

Reply:
Done in [3e33709](https://github.com/fleetdm/fleet/pull/21812/commits/3e33709ff3fcdc7dd9b19b05e8d42ac3dde97ccd).